src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java

```diff
@@ -963,7 +939,7 @@ private static boolean isPrimitive(DetailAST type) {
      */
     private static String getCanonicalName(DetailAST type) {
         final StringBuilder canonicalNameBuilder = new StringBuilder(256);
-        DetailAST toVisit = type.getFirstChild();
+        DetailAST toVisit = type;
```

Comment:
This is called from `visitImport` and `getTypeName`.

For `visitImport`, `type` is the the first child of the FQCN being imported, so its always the period. You can't import anything from the blank package, so we are always guaranteed a period first. This only looks at `IDENT` tokens, so regardless, it always skips over anything that is a period, and we can't import from the blank package so there is no way to kill it here. It is just slightly more processing to come to the same result.

For `getTypeName`, the `type` is always the `TYPE` token like in `VARIABLE_DEF`. As we said above, we only care about `IDENT` in this method, so we always skip `TYPE` for the next token.